pkg/mcp/sink/sink.go

```diff
@@ -183,31 +183,36 @@ func (sink *Sink) createInitialRequests() []*mcp.RequestResources {
 // stream interface and returns when a send or receive error occurs. The caller is responsible for
 // handling gRPC client/server specific error handling.
 func (sink *Sink) ProcessStream(stream Stream) error {
-	// send initial requests for each supported type
-	initialRequests := sink.createInitialRequests()
+	// send initial subscribe requests
+	go func() {
```

Comment:
This changes behavior. If the initial requests cannot be sent, the ProcessStream will still continue executing.

Reply:
I guess you mean this go func error eixt, and the belowing for loop will still run?

I think this can not happen, from my understanding if initial requests cannot be sent, the stream must be broken.

Reply:
Whatever we should separate send and receive in to different routines, as server side does.